### PR TITLE
Muc sub additional fuctions

### PR DESCRIPTION
**Pull request by CodeBast4rd**
Clean ups documentations. Also featured out a little the muc-sub plugin. But as the bug fixes and ng update should be already should be released this is the intermediate state.

### CODE REVIEW

File: projects/pazznetwork/ngx-chat/src/lib/services/adapters/xmpp/plugins/message-state.plugin.ts [hunk 1]
```diff
@@ -62,52 +60,56 @@ export class MessageStatePlugin extends AbstractXmppPlugin {
...
 
-    private async parseContactMessageStates() {
+    private async parseContactMessageStates(): Promise<void> {
         const itemElement = await this.publishSubscribePlugin.retrieveNodeItems(STORAGE_NGX_CHAT_CONTACT_MESSAGE_STATES);
         this.processPubSub(itemElement);
     }
 
-    private processPubSub(itemElement: Element[]) {
-        const results: JidToMessageStateDate = Object.create(null);
+    private processPubSub(itemElement: Stanza[]): void {
```
Comment (Frosty1604) [inline projects/pazznetwork/ngx-chat/src/lib/services/adapters/xmpp/plugins/message-state.plugin.ts:74]: Rename parameter `itemElement` to `itemElements`.

File: src/app/routes/index/index.component.ts [hunk 2]
```diff
@@ -20,14 +20,14 @@ import {
...
-    public unreadMessageCountPlugin: UnreadMessageCountPlugin;
-    public registrationMessage: string;
+    public domain?: string;
+    public service?: string;
+    public password?: string;
+    public username?: string;
+    public otherJid?: string;
+    public readonly multiUserChatPlugin: MultiUserChatPlugin;
+    public readonly unreadMessageCountPlugin: UnreadMessageCountPlugin;
+    public registrationMessage?: string;
```
Comment (Frosty1604) [inline src/app/routes/index/index.component.ts:30]: Remove `public` cause it is not needed.

**Review by Frosty1604 — APPROVED** [review]
lgtm

**Review by Frosty1604 — APPROVED** [review]
lgtm

File: projects/pazznetwork/ngx-chat/src/lib/services/adapters/xmpp/plugins/message-state.plugin.ts (hunk 1 above)
Comment (CodeBast4rd) [reply projects/pazznetwork/ngx-chat/src/lib/services/adapters/xmpp/plugins/message-state.plugin.ts]: Done

**Review by Frosty1604 — APPROVED** [review]

**Pull request merged**